### PR TITLE
several changes implied by new quartic fitting function

### PR DESCRIPTION
**Pull request by AndrewLevin**
The new quartic ARM DAC fitting function was not being printed out properly and there was no function to parse it.

## Description
Both of those things have been fixed.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
This is a follow up to https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/186 and preparation for https://github.com/cms-gem-daq-project/vfatqc-python-scripts/issues/223

## How Has This Been Tested?
The changes to `calibratedThrDac.py` has been tested.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: utils/anautilities.py [hunk 1]
```diff
@@ -875,6 +875,32 @@ def parseCalFile(filename=None):
 
     return (calDAC2Q_m, calDAC2Q_b)
 
+def parseArmDacCalFile(filename):
```
Comment (bdorney) [inline utils/anautilities.py:878]: Looks good except need function documentation here.  Please add then we can merge in.  See other functions in this file for examples.

**Review by bdorney — CHANGES_REQUESTED** [review]

File: utils/anautilities.py [hunk 2]
```diff
@@ -875,6 +875,32 @@ def parseCalFile(filename=None):
...
+
+    # Set the CAL DAC to fC conversion
+    coef4 = np.zeros(24)
+    coef3 = np.zeros(24)
+    coef2 = np.zeros(24)
+    coef1 = np.zeros(24)
+    coef0 = np.zeros(24)
+    list_bNames = ["vfatN","coef4","coef3","coef2","coef1","coef0"]
+    calTree = r.TTree('calTree','Tree holding VFAT Calibration Info')
+    calTree.ReadFile(filename)
```
Comment (bdorney) [inline utils/anautilities.py]: Also can you add this line `891` in a `try...except IOError` block with a `printRed` statement that says if `IOError` occurs the file is not readable or does not exist, e.g.:

```python
try:
    calTree.ReadFile(filename)
except IOError:
    printRed("file {0} is not readable or does not exist, please cross-check".format(filename))
    exit(os.EX_IOERR)
```

File: utils/anautilities.py (hunk 1 above)
Comment (AndrewLevin) [reply utils/anautilities.py:878]: done

File: utils/anautilities.py (hunk 2 above)
Comment (AndrewLevin) [reply utils/anautilities.py]: done

File: macros/calibrateThrDac.py [hunk 3]
```diff
@@ -505,18 +505,14 @@
             tgraph_scurveMeanVsThrDacForFit.SetPoint(tgraph_scurveMeanVsThrDacForFit.GetN(),thrDacVal,scurveMean)
             tgraph_scurveMeanVsThrDacForFit.SetPointError(tgraph_scurveMeanVsThrDacForFit.GetN()-1,0,scurveMeanError)
 
-        #a 4th order polynomial expanded about the lower edge of the fit range    
-        def quartic(x,par):    
-            return (par[0]*pow((x[0]-min(fitRange)),4)+par[1]*pow((x[0]-min(fitRange)),3))+par[2]*pow((x[0]-min(fitRange)),2)+par[3]*(x[0]-min(fitRange))+par[4]    
-
```
Comment (bdorney) [inline macros/calibrateThrDac.py:511]: Why was this removed?  It isn't described in the PR description.  What implications does this have on the output? E.g. the changes that you presented in the last SW dev's meeting.

Comment (AndrewLevin) [reply macros/calibrateThrDac.py:511]: in all of the tests I have done min(fitRange) = 0, so this has no effect

it just removes a feature that could have been used

I did this because including the min(fitRange) would add complication to the storing of the function in a text file or a database

File: utils/anautilities.py [hunk 4]
```diff
@@ -875,6 +875,56 @@ def parseCalFile(filename=None):
 
     return (calDAC2Q_m, calDAC2Q_b)
 
+def parseArmDacCalFile(filename):
+    """
+    Reads a text file and supplies the coefficients of the 
+    quartic polynomial used for the conversion between scurveMean 
+    (in fC) and CFG_THR_ARM_DAC.
```
Comment (bdorney) [inline utils/anautilities.py]: change lines 881-882 to:

> quartic polynomial used for converting between DAC units to charge units (in fC) for CFG_THR_ARM_DAC

File: macros/calibrateThrDac.py (hunk 3 above)
Comment (bdorney) [reply macros/calibrateThrDac.py:511]: Okay thanks for clarifying.

File: utils/anautilities.py [hunk 5]
```diff
@@ -875,6 +875,56 @@ def parseCalFile(filename=None):
...
+    coef3 = np.zeros(24)
+    coef2 = np.zeros(24)
+    coef1 = np.zeros(24)
+    coef0 = np.zeros(24)
+    list_bNames = ["vfatN","coef4","coef3","coef2","coef1","coef0"]
+    calTree = r.TTree('calTree','Tree holding VFAT Calibration Info')
+    try:
+        calTree.ReadFile(filename)
+    except IOError:
+        printRed("file {0} is not readable or does not exist, please cross-check".format(filename))
```
Comment (bdorney) [inline utils/anautilities.py:914]: I didn't check if the `printRed` function needs to be imported or not.  Please cross check that this is already taken care of or add the appropriate import.

Comment (AndrewLevin) [reply utils/anautilities.py:914]: fixed

File: utils/anautilities.py (hunk 4 above)
Comment (AndrewLevin) [reply utils/anautilities.py]: done

**Review by bdorney — APPROVED** [review]
